stm32CubeProg.sh

```diff
@@ -53,11 +55,69 @@ aborting() {
   exit 1
 }
 
+# Check STM32CubeProgrammer cli availability and getopt version
+case "${UNAME_OS}" in
+  Linux*)
+    STM32CP_CLI=STM32_Programmer.sh
+    if ! command -v $STM32CP_CLI >/dev/null 2>&1; then
+      export PATH="$HOME/STMicroelectronics/STM32Cube/STM32CubeProgrammer/bin":"$PATH"
+    fi
+    if ! command -v $STM32CP_CLI >/dev/null 2>&1; then
+      export PATH="/opt/stm32cubeprog/bin":"$PATH"
+    fi
+    if ! command -v $STM32CP_CLI >/dev/null 2>&1; then
+      aborting
+    fi
+    ;;
+  Darwin*)
+    STM32CP_CLI=STM32_Programmer_CLI
+    if ! command -v $STM32CP_CLI >/dev/null 2>&1; then
+      export PATH="/Applications/STMicroelectronics/STM32Cube/STM32CubeProgrammer/STM32CubeProgrammer.app/Contents/MacOs/bin":"$PATH"
+    fi
+    if ! command -v $STM32CP_CLI >/dev/null 2>&1; then
+      aborting
+    fi
+    if ! command -v /usr/local/opt/gnu-getopt/bin/getopt >/dev/null 2>&1; then
```

Comment:
By the way, for Apple Silicon devices, the path should be `/opt/homebrew/opt/gnu-getopt/`, maybe this needs to be checked as well?

Reply:
Hi @lhvy 
Thanks for the feedback, I'm not familiar with mac.
So something like this:
```shell
    if ! command -v /usr/local/opt/gnu-getopt/bin/getopt >/dev/null 2>&1; then
      if ! command -v /opt/homebrew/opt/gnu-getopt/bin/getopt >/dev/null 2>&1; then
        echo "Warning: long options not supported due to getopt from FreeBSD usage."
        GNU_GETOPT=n
      else
        export PATH="/opt/homebrew/opt/gnu-getopt/bin":"$PATH"
      fi
    else
      export PATH="/usr/local/opt/gnu-getopt/bin":"$PATH"
    fi
```